source/request.js

```diff
@@ -181,6 +181,13 @@ define(['airports'], function (airports) {
     return this.getCloudFlareLocationCode();
   };
 
+  Request.prototype.getCloudFlareTrace = function () {
```

Comment:
Extensions should have access to the browser's `URL` object. This should probably be done in terms of that, rather than splitting and parsing strings.